src/Components/bookHub/BookCard.js

```diff
@@ -0,0 +1,60 @@
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import React, { useState, useEffect } from 'react';
+
+import './BookCard.css';
+
+// airtable configuration
+const Airtable = require('airtable');
+
+const airtableConfig = {
+  apiKey: process.env.REACT_APP_AIRTABLE_USER_KEY,
+  baseKey: process.env.REACT_APP_AIRTABLE_BASE_KEY,
+};
+
+const base = new Airtable({ apiKey: airtableConfig.apiKey })
+  .base(airtableConfig.baseKey);
+
+export default function Card({
+  id, title, author, image,
+}) {
+  const [authorVar, setAuthor] = useState();
+
+  const authorRecords = () => {
+    base('Creator').find(author, (err, record) => {
+      if (err) { console.error(err); }
+      setAuthor(record);
+    });
+  };
+
+  useEffect(authorRecords, []);
+
+  return ( // horizontal scroll not implemented
+    <div className="card">
+      <Link to={`/book/${id}`}>Book Page</Link>
+      <div className="image">
+        <img className="book-cover" src={image} alt="missing_book_cover" />
+      </div>
+      <div className="title">
+        {' '}
+        {title}
+      </div>
+      <div className="author">
+        By
+        {' '}
+        { authorVar !== undefined ? (authorVar.fields.name) : 'loading'}
```

Comment:
This line is the reason you see 'loading' instead of 'MISSING AUTHOR'. authorVar holds the results of the airtable call, so if the authorID is set to 'MISSING AUTHOR' as you've specified, then authorVar will just remain undefined, and 'loading' will be displayed.

If you want to display 'MISSING AUTHOR' you could do this
```suggestion
        { authorVar !== undefined ? (authorVar.fields.name) : author}
```
as the author prop will hold the message you are expecting. But the way you have coded this is fine as is!

It would be best to add a check within the getAuthor function so that you don't make the airtable call at all if you know it will fail.

Reply:
Ok yes changing it works, thank you! I have another very minor thing, it's that when the author field is not filled, it will display "By MISSING AUTHOR". Is there anyway to change that to just "MISSING AUTHOR"? It is from line 43 to 45.

Reply:
Yeah! If you look at how you've coded this section, "By" is outside the conditional so it will be displayed regardless! I would deletelines 43 and 44 and then change the conditional to this:

```suggestion
        { authorVar !== undefined ? 
          "By ${authorVar.fields.name}":  "MISSING AUTHOR"}
```

That way if the author is defined,  "By [name]" will be displayed, and otherwise the error message alone without the 'Missing Author' will be shown.

You'll notice that this version doesn't just show the error message stored in the author prop as the alternative. It has the error message written out. I think that this is a bit clearer to read, so we know an error message would be displayed. Otherwise a person who doesn't have context might be confused as to what 'author' holds.